tests/fsharp/Compiler/Language/StringInterpolation.fs

```diff
@@ -0,0 +1,713 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.UnitTests
+
+open NUnit.Framework
+open FSharp.Compiler.SourceCodeServices
+open FSharp.Test.Utilities
+
+[<TestFixture>]
+module StringInterpolationTests =
+
+    [<Test>]
+    let ``Basic string interpolation`` () =
+        CompilerAssert.CompileExeAndRunWithOptions [| "--langversion:preview" |]
+            """
+let check msg a b = 
+    if a = b then printfn "%s succeeded" msg else failwithf "%s failed, expected %A, got %A" msg b a
+
+check "vcewweh1" $"this is 2" "this is 2"
```

Comment:
I think this is made so that multiple "assertions" are packed into a single compiler run.
Inside the run the test framework is not accessible so we have to build a mechanism to throw the errors out.

The cryptographical name, if laid out uniquely, can help to identify the location of the error without reading and understanding the title of the test.

Reply:
> ...Each of these should be a distinct...

Yes this is too much test explosion and churn for me 

1. We already use the technique to raise exceptions in generated code in order to check evaluation results in both this test suite (Slicing tests) and in `fsharp\core` test suite.  I can't particularly see a better technique, especially for string interpolation where the easiest way to test is to compare an interpolated string against a non-interpolated string

2. Repeating the same exception-injection code on each of 200 test cases already in this file in a separate test case for each one would create like 7000 line monster.

3. Factoring out the exception injection code is possible but just becomes increasingly baroque, it's no improvement

4. Regarding naming of test cases like `vcewweh1` - the technique of giving a random gibberish identifier to each individual test has worked well for years, it's easily searchable and identifiable.  Giving individual meaningful names in this case to every micro test would be pretty explosive.  It's very common to integrate multiple micro tests into one test case e.g. https://github.com/DiffSharp/DiffSharp/blob/dev/tests/DiffSharp.Tests/TestTensor.fs#L5626 to choose a random example

Anyway let's focus on the logical content of the tests and the completeness of the test matrix and how we make that more evident.